neuraxle/data_container.py

```diff
@@ -113,12 +113,19 @@ def convolved_1d(self, stride, kernel_size) -> Iterable['DataContainer']:
         .. seealso::
             `<https://github.com/guillaume-chevalier/python-conv-lib>`_
         """
-        conv_current_ids = convolved_1d(stride=stride, iterable=self.current_ids, kernel_size=kernel_size)
-        conv_data_inputs = convolved_1d(stride=stride, iterable=self.data_inputs, kernel_size=kernel_size)
-        conv_expected_outputs = convolved_1d(stride=stride, iterable=self.expected_outputs, kernel_size=kernel_size)
+        conv_current_ids = convolved_1d(stride=stride, iterable=self.current_ids, kernel_size=kernel_size, include_incomplete_pass=True)
+        conv_data_inputs = convolved_1d(stride=stride, iterable=self.data_inputs, kernel_size=kernel_size, include_incomplete_pass=True)
+        conv_expected_outputs = convolved_1d(stride=stride, iterable=self.expected_outputs, kernel_size=kernel_size, include_incomplete_pass=True)
 
         for current_ids, data_inputs, expected_outputs in zip(conv_current_ids, conv_data_inputs,
                                                               conv_expected_outputs):
+            for i, (ci, di, eo) in enumerate(zip(current_ids, data_inputs, expected_outputs)):
+                if di is None:
+                    current_ids = current_ids[:i]
+                    data_inputs = data_inputs[:i]
+                    expected_outputs = expected_outputs[:i]
```

Comment:
Are you sure, `[:i]` ? This looks like it would introduce a bug to me, because the slice gets everything from the begginning and it's not using the batch size "`kernel_size`". Did you mean `[i*kernel_size:(i+1)*kernel_size]`?

Reply:
conv creates None at the end of an incomplete pass... so if the kernel size is 10, and we make a slice of 10 elements starting from the last 5 elements it will return the last 5 elements followed by None values. this code makes me only return the values that are not None. is this ok ? @guillaume-chevalier

Reply:
Ok, so you are saying that the fix in conv==0.2 would need another flag which is something like `don't zero pad`?

Reply:
@guillaume-chevalier yes !

Reply:
Resolving to issue.